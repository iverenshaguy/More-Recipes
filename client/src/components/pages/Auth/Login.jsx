import React from 'react';
import PropTypes from 'prop-types';
import Form from '../../shared/Forms';

/**
 * @exports
 * @function Login
 * @param {object} props
 * @returns {JSX} Login
 */
const Login = (props) => {
  const loginMeta = {
    title: 'Sign In to Your Account',
    btnText: 'SIGN IN',
    extra: <p className="text-center">{'Don\'t have an account, '}<a href="/signup" onClick={e => props.changeForm(e, 'signup')}>Sign up here.</a></p>
  };

  return <Form {...props} type="login" meta={loginMeta} changeForm={props.changeForm} />;
};

Login.propTypes = {
  changeForm: PropTypes.func.isRequired
};

export default Login;

