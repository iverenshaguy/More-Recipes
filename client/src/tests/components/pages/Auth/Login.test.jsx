import React from 'react';
import Login from '../../../../components/pages/Auth/Login';

const changeFormMock = jest.fn();


describe('Login', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<Login changeForm={changeFormMock} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
