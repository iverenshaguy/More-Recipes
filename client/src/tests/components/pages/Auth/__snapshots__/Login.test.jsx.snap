// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<Connect(Form)
  changeForm={[Function]}
  meta={
    Object {
      "btnText": "SIGN IN",
      "extra": <p
        className="text-center"
      >
        Don't have an account, 
        <a
          href="/signup"
          onClick={[Function]}
        >
          Sign up here.
        </a>
      </p>,
      "title": "Sign In to Your Account",
    }
  }
  type="login"
/>
`;
